testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java

```diff
@@ -0,0 +1,678 @@
+            FileUtils.deleteQuietly(rolesAuthzRealmFile);
+            FileUtils.deleteDirectory(new File(fsRealmPath));
+            FileUtils.deleteDirectory(tempFolder);
+        }
+
+        private void addUserToFilesystemRealm(CLIWrapper cli, String username, String password) throws Exception {
+            addUserToFilesystemRealm(cli, username, password, null);
+        }
+
+        private void addUserToFilesystemRealm(CLIWrapper cli, String username, String password, String role)
+                throws Exception {
+            cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s/identity=%s:add()",
+                    FILESYSTEM_REALM_AUTHN_NAME, username));
+            cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s/identity=%s:set-password(clear={password=\"%s\"})",
+                    FILESYSTEM_REALM_AUTHN_NAME, username, password));
+            if (role != null) {
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s/identity=%s:add-attribute(name=Roles, value=[\"%s\"])",
+                        FILESYSTEM_REALM_AUTHN_NAME, username, role));
+            }
+        }
+
+        private void addAggregateRealmAndRelatedResources(CLIWrapper cli, String name, String authnRealm, String authzRealm) {
+            cli.sendLine(String.format(
+                    "/subsystem=elytron/aggregate-realm=%s:add(authentication-realm=%s,authorization-realm=%s)",
+                    name, authnRealm, authzRealm));
+            cli.sendLine(String.format(
+                    "/subsystem=elytron/security-domain=%1$s:add(realms=[{realm=%1$s,role-decoder=groups-to-roles}],default-realm=%1$s,permission-mapper=default-permission-mapper)",
+                    name));
+            cli.sendLine(String.format(
+                    "/subsystem=elytron/http-authentication-factory=%1$s:add(http-server-mechanism-factory=%2$s,security-domain=%1$s,"
+                    + "mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=\"Some realm\"}]}])",
+                    name, PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY));
+            cli.sendLine(String.format(
+                    "/subsystem=undertow/application-security-domain=%1$s:add(http-authentication-factory=%1$s)",
+                    name));
+        }
+
+        private void removeAggregateRealmAndRelatedResources(ManagementClient mc, CLIWrapper cli, String name) throws Exception {
+            cli.sendLine(String.format("/subsystem=undertow/application-security-domain=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/http-authentication-factory=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/aggregate-realm=%s:remove()", name));
+            ServerReload.reloadIfRequired(mc.getControllerClient());
```

Comment:
Move this call just after the 2nd call of `removeAggregateRealmAndRelatedResources` method in the `tearDown`. IMO, it's not necessary to do it for every single aggregate realm.